### PR TITLE
Improve virtual touch target positioning for narrow elements

### PR DESCRIPTION
**Pull request by michaeltaranto**
Improve virtual touch target positioning for narrow elements

To maintain accessibility for smaller interactive elements, Braid uses a virtual touch target to maintain the minimum hit area This change ensures that the virtual element is always centered to the visual target, in particular when the width of the visual target is narrower than the minimum hit area.

### Dev note
Moved style composition of virtual touch targets into Vanilla Extract style sheets as a minor optimisation.

### CODE REVIEW

File: packages/braid-design-system/src/lib/components/Accordion/AccordionItem.css.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import { style } from '@vanilla-extract/css';
 import { calc } from '@vanilla-extract/css-utils';
 import { vars } from '../../themes/vars.css';
+import { virtualTouchable } from '../private/touchable/virtualTouchable.css';
 
-export const button = style({});
+export const button = style([{}, virtualTouchable]);
```
Comment (askoufis) [inline packages/braid-design-system/src/lib/components/Accordion/AccordionItem.css.ts:6]: Any reason not to just re-export the classname?
```suggestion
export const button = virtualTouchable;
```

Comment (michaeltaranto) [reply packages/braid-design-system/src/lib/components/Accordion/AccordionItem.css.ts:6]: Yep need to create a new reference to be used in the following selectors

File: packages/braid-design-system/src/lib/components/private/touchable/virtualTouchableRules.ts [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+*/
 export const virtualTouchableRules = {
-  transform: 'translateY(-50%)',
   minHeight: hitArea,
   minWidth: hitArea,
   height: '100%',
+  width: '100%',
+  transform: 'translate(-50%, -50%)',
   top: '50%',
+  left: '50%',
```
Comment (askoufis) [inline packages/braid-design-system/src/lib/components/private/touchable/virtualTouchableRules.ts:14]: Do we have/want a screenshot test for this?

Comment (michaeltaranto) [reply packages/braid-design-system/src/lib/components/private/touchable/virtualTouchableRules.ts:14]: Thats an interesting one actually. Might have a look at the idea of having a screenshot of the implementation in the components themselves.

Comment (michaeltaranto) [reply packages/braid-design-system/src/lib/components/private/touchable/virtualTouchableRules.ts:14]: Something like this:

Screenshot 2024-05-16 at 1 55 04 PM

Comment (askoufis) [reply packages/braid-design-system/src/lib/components/private/touchable/virtualTouchableRules.ts:14]: I like it 👍

Comment (askoufis) [reply packages/braid-design-system/src/lib/components/private/touchable/virtualTouchableRules.ts:14]: Worth doing in this PR, or separately?

Comment (michaeltaranto) [reply packages/braid-design-system/src/lib/components/private/touchable/virtualTouchableRules.ts:14]: I'll add to this one if you're happy?

Comment (askoufis) [reply packages/braid-design-system/src/lib/components/private/touchable/virtualTouchableRules.ts:14]: Go for it.

File: packages/braid-design-system/src/lib/components/private/touchable/debugTouchable.ts [hunk 3]
```diff
@@ -1,13 +1,12 @@
 import type { StyleRule } from '@vanilla-extract/css';
 
-export const debugTouchable = ({ after = false } = {}): StyleRule =>
-  process.env.NODE_ENV === 'production'
```
Comment (michaeltaranto) [inline packages/braid-design-system/src/lib/components/private/touchable/debugTouchable.ts:4]: Not sold on removing this check.

### Pro of stripping the style
- Any accidental debug attributes in code dont cause red boxes for our users

### Cons
- Cant debug in Playroom or Storybook without further work

Comment (askoufis) [reply packages/braid-design-system/src/lib/components/private/touchable/debugTouchable.ts:4]: You could invert the condition and change the name of the envar to be something other than `NODE_ENV`, and only set it for playroom/storybook.
E.g. `process.env.DEBUG_TOUCHABLE ? {debugStyle} : {}`

File: .changeset/yellow-badgers-cover.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  - Radio
+  - TextDropdown
+  - TextLink
+  - TextLinkButton
+  - Toggle
+---
+
+Improve virtual touch target positioning for narrow elements
+
+To maintain accessibility for smaller interactive elements, Braid uses a virtual touch target to maintain the minimum hit area
```
Comment (askoufis) [inline .changeset/yellow-badgers-cover.md:24]:
```suggestion
To maintain accessibility for smaller interactive elements, Braid uses a virtual touch target to maintain the minimum hit area.
```

File: .changeset/twelve-dryers-swim.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+---
+
+---
+updated:
+  - MenuButton
+---
+
+**MenuButton:** Improve virtual touch target positioning for narrow elements
+
+To maintain accessibility for smaller interactive elements, Braid uses a virtual touch target to maintain the minimum hit area
```
Comment (askoufis) [inline .changeset/twelve-dryers-swim.md:12]:
```suggestion
To maintain accessibility for smaller interactive elements, Braid uses a virtual touch target to maintain the minimum hit area.
```

**Review by askoufis — APPROVED** [review]

**Pull request merged**